### PR TITLE
adding sort by usage feature

### PR DESCRIPTION
**Pull request by lahiru823**
Please review and commit

### CODE REVIEW

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 1]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+        }
+        else if (paging.sort == 'usage') {
+            server.privileged(function () {
+                var config = require('/config/store.json');
+                var userName = user ? user.username : "";
+                var index = 0, maxTry = 0;
+                var getNextassetSet = function () {
+                    var bamService = Packages.org.wso2.carbon.appmgt.impl.dao.ApiMgtDAO;
+                    var bamServiceObj = new bamService();
+                    log.debug("Object initialized successfully : org.wso2.carbon.appmgt.impl.dao.ApiMgtDAO ");
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Check isDebugEnabled();

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 2]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+
+                    var result = bamServiceObj.getAppsByHitCount(userName, (paging.start + index), 12, config.uiActivityPublishDataSource);
```
Comment (rushmin) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: 12 is a magic number. Make its a constant (or a variable)

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 3]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+                    log.debug("query executed successfully to get the assets hit count");
+
+                    index += 12;
+                    return JSON.parse(result);
+                };
+
+                assets = [];
+                var aid, asset;
+                while (assets.length < 12 && maxTry < 10) {
```
Comment (rushmin) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: 10 is a magic number. Make it a constant (or a variable)

Comment (rushmin) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: maxTry is not a meaningful name in this case. It should be something like 'currentTry'

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 4]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+                    maxTry++;
+                    var result = getNextassetSet();
+                    for (var n = 0; n < result.length && assets.length < 12; n++) {
```
Comment (rushmin) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: checking result.length for not null is redundant.

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 5]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
                 }
             });
 
-        } else {
+        }
+        else if (paging.sort == 'usage') {
```
Comment (rushmin) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Add a comment to explain the logic.

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 6]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+
+                    var result = bamServiceObj.getAppsByHitCount(userName, (paging.start + index), 12, config.uiActivityPublishDataSource);
+                    log.debug("query executed successfully to get the assets hit count");
+
+                    index += 12;
+                    return JSON.parse(result);
+                };
+
+                assets = [];
+                var aid, asset;
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Use meaningful variable names. "aid"

File: modules/apps/store/extensions/assets/webapp/pages/assets.jag [hunk 7]
```diff
@@ -96,7 +113,45 @@ require('/modules/store.js').exec(function (ctx) {
                 }
             });
 
-        } else {
+        }
+        else if (paging.sort == 'usage') {
```
Comment (rushmin) [inline modules/apps/store/extensions/assets/webapp/pages/assets.jag]: Please do the same changes which where mentioned in the same logic of the mobile apps.

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 8]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+                    for (var n = 0; n < result.length && assets.length < 12; n++) {
+                        aid = String(result[n].UUID);
+                        try {
+                            asset = store.asset(type, aid);
+                            asset.indashboard = store.isuserasset(aid, type);
+                            if (configs.lifeCycleBehaviour.visibleIn.indexOf(String(asset.lifecycleState), 0) >= 0) {
+                                assets.push(asset);
+                            }
+                        } catch (e) {
+                            log.warn("error retrieving asset from store ,information might be stale in table. id=" +
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Capitalize the sentence first letter.

File: modules/apps/store/themes/store/pages/2-column-left.hbs [hunk 9]
```diff
@@ -69,6 +69,6 @@
 <script src="{{url "/themes/store/js/caramel-client.js"}}"></script>
 <script src="{{url "/themes/store/js/jquery.noty.packaged.min.js"}}"></script>
 <script src="{{url "/themes/store/js/notifications.js"}}"></script>
-
+    <script src="{{url "/themes/store/js/writeToPublisher.js"}}"></script>
```
Comment (rushmin) [inline modules/apps/store/themes/store/pages/2-column-left.hbs]: Spacing issue.

File: modules/apps/store/themes/store/pages/1-column.hbs [hunk 10]
```diff
@@ -64,5 +64,6 @@
 <script src="{{url "/themes/store/js/theme.js"}}"></script>
     {{js .}}
 <script src="{{url "/themes/store/js/caramel-client.js"}}"></script>
+    <script src="{{url "/themes/store/js/writeToPublisher.js"}}"></script>
```
Comment (rushmin) [inline modules/apps/store/themes/store/pages/1-column.hbs]: Spacing issue

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 1 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: this is already fixed in a sub sequent PR

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 5 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: commit# 3267c526db8f770622ed4370a2876daa72cba4ac

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 2 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: fixed with commit# 27065f221d16b236085fac76d72aa94cbc021468

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 6 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: fixed with commit# 75f74c8249d5af07cac477af29561fba80652d20

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 3 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: fixed with commit# b8127c2e754da7ad8521f766e98d6dfbccbbe98b

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 8 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: fixed with commit# b79f58cf8347d5699cc3289ecbef5f9f7626992f

File: modules/apps/store/extensions/assets/webapp/pages/assets.jag (hunk 7 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/webapp/pages/assets.jag]: fixed with commit# d9c883a37eb09944a615a77dd4942af2d4171fce

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 4 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: fixed with commit# 30ba7b3f2127f7f577673a907560b997563ef86d

File: modules/apps/store/themes/store/pages/1-column.hbs (hunk 10 above)
Comment (lahiru823) [inline modules/apps/store/themes/store/pages/1-column.hbs]: already fixed with commit# cde1b9020dd61e218e832fd31e0b3cd565a57124

File: modules/apps/store/themes/store/pages/2-column-left.hbs (hunk 9 above)
Comment (lahiru823) [inline modules/apps/store/themes/store/pages/2-column-left.hbs]: this js reference has been removed with a subsequent commit

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 11]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+                assets = [];
+                var aid, asset;
+                while (assets.length < 12 && maxTry < 10) {
+                    maxTry++;
+                    var result = getNextassetSet();
+                    for (var n = 0; n < result.length && assets.length < 12; n++) {
+                        aid = String(result[n].UUID);
+                        try {
+                            asset = store.asset(type, aid);
+                            asset.indashboard = store.isuserasset(aid, type);
```
Comment (manuranga) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: better to use camelCase

File: modules/apps/store/themes/store/js/sort-assets.js [hunk 12]
```diff
@@ -4,8 +4,7 @@ var updateSortUI = function() {
 
 $(function() {
 	updateSortUI();
-
-	$('#breadcrumb > div').live('click', function(e) {
+		$('#breadcrumb > div').live('click', function(e) {
```
Comment (manuranga) [inline modules/apps/store/themes/store/js/sort-assets.js]: formatting issue here

**Pull request closed without merging**